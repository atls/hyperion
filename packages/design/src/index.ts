export * from './Sample'
export * from './Layout'
export * from './Container'
export * from './List'
export * from './Text'
export * from './Space'
