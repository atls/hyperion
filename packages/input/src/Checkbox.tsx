import React  from 'react'

import styled from '@emotion/styled'

const Container = styled.label(({ theme }: any) => ({
  display: 'flex',
  position: 'relative',
  paddingLeft: 23,
  cursor: 'pointer',
  userSelect: 'none',
  '&:hover input ~ div': {
    backgroundColor: theme.colors.mountainmist,
  },
  '& input:checked ~ div': {
    backgroundColor: theme.colors.mountainmist,
  },
  '& input:checked ~ div:after': {
    display: 'block',
  },
}))

const Input = styled.input({
  position: 'absolute',
  opacity: 0,
  cursor: 'pointer',
  height: 0,
  width: 0,
})

const CheckMark = styled.div(
  ({
    theme,
    width,
    height,
    borderRadius,
    markColor,
    markBorderWidth,
    markWidth,
    markHeight,
  }: any) => ({
    position: 'absolute',
    top: 0,
    left: 0,
    width: width || 18,
    height: height || 18,
    boxSizing: 'border-box',
    borderRadius: theme.radii[borderRadius] || theme.radii.small,
    backgroundColor: theme.colors.alto,
    '&:after': {
      content: '""',
      position: 'absolute',
      display: 'none',
      left: 6,
      top: 1,
      width: markWidth || 5,
      height: markHeight || 10,
      border: (markColor && `solid ${theme.colors[markColor]}`) || 'solid white',
      borderWidth: (markWidth && `0 ${markBorderWidth} ${markBorderWidth} 0`) || '0 2px 2px 0',
      transform: 'rotate(45deg)',
    },
  }),
)

export const Checkbox = ({ children, ...props }) => (
  <Container {...props}>
    {children}
    <Input
      type='checkbox'
      onChange={({ target }) => props.onChange && props.onChange(target.value)}
      {...props}
    />
    <CheckMark {...props} />
  </Container>
)
