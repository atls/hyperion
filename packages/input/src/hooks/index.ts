import { useAttach }     from './useAttach'
import { useMask }       from './useMask'
import { useOnChange }   from './useOnChange'
import { useOnKeyPress } from './useOnKeyPress'

const compose = (...items) => props => Object.assign({}, ...items.map(item => item(props)))

export const useHooks = compose(useOnKeyPress, useOnChange, useAttach, useMask)
