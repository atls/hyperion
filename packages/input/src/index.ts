export * from './Textarea'
export * from './Input'
export * from './Checkbox'
