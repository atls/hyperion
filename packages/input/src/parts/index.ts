export * from './StyledInput'
export * from './PrefixSuffix'
export * from './Addon'

export * from './Attachments'
export * from './Addons'
