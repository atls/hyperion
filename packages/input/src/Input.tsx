import React                                                from 'react'

import { Addons, Attachments, LabelContainer, StyledInput } from './parts'
import { useHooks }                                         from './hooks'

const Input = (props: any) => {
  const hooksProps = useHooks(props)
  const finalProps = { ...props, ...hooksProps }

  return (
    <Addons {...finalProps}>
      <Attachments {...finalProps}>
        <LabelContainer {...finalProps}>
          <StyledInput {...finalProps} />
        </LabelContainer>
      </Attachments>
    </Addons>
  )
}

Input.defaultProps = {
  size: 'normal',
}

export { Input }
