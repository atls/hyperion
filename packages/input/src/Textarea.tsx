import React           from 'react'
import { boxShadow }   from 'styled-system'

import styled          from '@emotion/styled'

import { StyledInput } from './parts'

const Input = (props) => (
  <StyledInput
    {...props}
    onChange={({ target }) => props.onChange && props.onChange(target.value)}
  />
)

const Textarea = styled(Input)(
  () => ({
    padding: '17px 20px',
    resize: 'none',
    height: 'auto',
  }),
  boxShadow,
)

Textarea.defaultProps = {
  ...StyledInput.defaultProps,
  as: 'textarea',
  size: 'normal',
}

export { Textarea }
