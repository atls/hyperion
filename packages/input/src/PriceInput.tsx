import React            from 'react'
import createNumberMask from 'text-mask-addons/dist/createNumberMask'

import { Input }        from './Input'

const mask = createNumberMask({
  prefix: '',
  suffix: ' ₽',
  thousandsSeparatorSymbol: ' ',
})

export const PriceInput = ({ onChange = (a) => {}, ...props }) => {
  const onChangeFormatted = (value) => {
    onChange(typeof value === 'string' ? Number(value.replace(/\s|₽/g, '')) : value)
  }

  return <Input onChange={onChangeFormatted} mask={mask} {...props} />
}
