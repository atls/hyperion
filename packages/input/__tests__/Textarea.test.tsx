import React             from 'react'
import renderer          from 'react-test-renderer'
import { ThemeProvider } from 'emotion-theming'

import { Textarea }      from '../src/index'

it('Textarea should render correctly', () => {
  const defaultTheme = {
    colors: {
      green: '#0f0',
      black: '#000',
      white: '#fff',
      blue: '#00f',
      red: '#f00',
      yellow: 'yellow',
      lightGray: 'lightGray',
      silver: '#C0C0C0',
      mountainmist: '#ecece3',
      alto: '#dbdbdb',
    },
    fontSizes: {
      tiny: '5em',
      normal: '7em',
      regular: '9em',
    },
    shadows: {
      controls: '0 0 10px rgba(0,0,0,0.5)',
    },
    borders: {
      controls: '2px solid black',
    },
    radii: {
      small: '5px',
    },
  }

  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <Textarea placeholder='Placeholder' border='1px solid black' />
      </ThemeProvider>,
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
