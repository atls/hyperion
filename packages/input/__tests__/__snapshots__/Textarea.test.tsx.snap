// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  outline: none;
  border: none;
  width: 100%;
  font-family: primary;
  font-weight: medium;
  border: 1px solid black;
  border-radius: extra;
  color: #000;
  background-color: #fff;
  padding: 17px 20px;
  resize: none;
  height: auto;
}

.emotion-0::-webkit-input-placeholder {
  color: #C0C0C0;
}

.emotion-0::-moz-placeholder {
  color: #C0C0C0;
}

.emotion-0:-ms-input-placeholder {
  color: #C0C0C0;
}

.emotion-0::placeholder {
  color: #C0C0C0;
}

<textarea
  className="emotion-0"
  onChange={[Function]}
  placeholder="Placeholder"
  size="normal"
/>
`;
